src/main/java/mate/academy/springboot/datajpa/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.springboot.datajpa.service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.repository.specification.ProductSpecificationManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final ProductSpecificationManager productSpecificationManager;
+
+    public ProductServiceImpl(ProductRepository productRepository,
+                              ProductSpecificationManager productSpecificationManager) {
+        this.productRepository = productRepository;
+        this.productSpecificationManager = productSpecificationManager;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product getById(Long id) {
+        return productRepository.findById(id).orElseThrow(NoSuchElementException::new);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public List<Product> getAllByPriceBetween(BigDecimal from, BigDecimal to) {
+        return productRepository.findProductsByPriceBetween(from, to);
+    }
+
+    @Override
+    public List<Product> getAll(Map<String, String> params) {
+        Specification<Product> specification = null;
+        for (Map.Entry<String, String> entry : params.entrySet()) {
+            Specification<Product> specific = productSpecificationManager.get(entry.getKey(),
+                    entry.getValue().split(", "));
```

Comment:
```suggestion
                    entry.getValue().split(","));
```
URL can’t  have whitespaces